### ISSUE TITLE
Hash-bug in ReturnnSearchJob

### ISSUE THREAD

**Issue by JackTemaki** [rwth-i6/i6_core#274]
#63 refactored the hashing for the `ReturnnSearchJob`, but I just noticed during hash tests for a completely unrelated pipeline that here was a flawed hashing introduced that nobody noticed. `Checkpoint.ckpt_path` is called during hash computation, which calls a `.get_path()`. Unfortunately, `cpkt_path` was not implemented with `DelayedSlice` at that time, and kept to not break old code too much. I now see that this was not the best idea to keep the compatibility while we were still in the "unstable" period.

All we would need to do is changing:
```
        config = {
            "load": model_checkpoint.ckpt_path,
            "search_output_file_format": output_mode,
            "need_data": False,
            "search_do_eval": 0,
        }
```
To 
```
        config = {
            "load": model_checkpoint,
            "search_output_file_format": output_mode,
            "need_data": False,
            "search_do_eval": 0,
        }
```

So I guess the only chance is to introduce a `ReturnnSearchJobV2`, because in this case expecting a graph migration from all users is not suitable anymore. I guess we can use this point to also refactor other parts of the ReturnnSearchJob.

Opinions please, as this will be the case where we define what will happen in the future when such things occur.

**Comment by christophmluscher** [comment]
I prefer the first approach as well.

I find it hard to estimate  how much the `ReturnnSearchJob` is used and therefor how much people would be affected by breaking the hash..

**Comment by JackTemaki** [comment]
Resolved via #278
